### PR TITLE
Deprecate the wiki and remove the lint list from the README (fixes #1933)

### LINKED ISSUE #1933: Make gh-pages "the one" documentation
Currently we have the [gh-pages](https://rust-lang-nursery.github.io/rust-clippy/master/index.html), [the wiki](https://github.com/rust-lang-nursery/rust-clippy/wiki) and the [lint list in the README](https://github.com/rust-lang-nursery/rust-clippy/blob/master/README.md#lints).

I suggest we just use the gh-pages (they are so much prettier and better to use) and modify our links to point there (and make the README also just point there).

The wiki would get one last update with a big red banner on top pointing to the gh-pages.

Another bonus: the gh-pages are autogenerated, while the wiki requires manual publishing

### LINKED ISSUE #2005: No wiki entry for for cast_lossless?
The cast_lossless lint references https://github.com/rust-lang-nursery/rust-clippy/wiki#cast_lossless - however there appears to be no entry for this lint.

### PR DESCRIPTION
**Pull request by oli-obk**
fixes #2005

### CODE REVIEW

File: clippy_lints/src/utils/mod.rs [hunk 1]
```diff
@@ -566,7 +566,7 @@ impl<'a> DiagnosticWrapper<'a> {
     fn wiki_link(&mut self, lint: &'static Lint) {
         if env::var("CLIPPY_DISABLE_WIKI_LINKS").is_err() {
             self.0.help(&format!(
-                "for further information visit https://github.com/rust-lang-nursery/rust-clippy/wiki#{}",
+                "for further information visit https://rust-lang-nursery.github.io/rust-clippy/master/index.html#{}",
```
Comment (mcarton) [inline clippy_lints/src/utils/mod.rs]: This should not point to master but the current package version `CARGO_PKG_VERSION`.

File: README.md [hunk 2]
```diff
@@ -181,218 +181,7 @@ transparently:
...
-[wrong_transmute](https://github.com/rust-lang-nursery/rust-clippy/wiki#wrong_transmute)                                     | warn    | transmutes that are confusing at best, undefined behaviour at worst and always useless
-[zero_divided_by_zero](https://github.com/rust-lang-nursery/rust-clippy/wiki#zero_divided_by_zero)                           | warn    | usage of `0.0 / 0.0` to obtain NaN instead of std::f32::NaN or std::f64::NaN
-[zero_prefixed_literal](https://github.com/rust-lang-nursery/rust-clippy/wiki#zero_prefixed_literal)                         | warn    | integer literals starting with `0`
-[zero_ptr](https://github.com/rust-lang-nursery/rust-clippy/wiki#zero_ptr)                                                   | warn    | using 0 as *{const, mut} T
-[zero_width_space](https://github.com/rust-lang-nursery/rust-clippy/wiki#zero_width_space)                                   | deny    | using a zero-width space in a string literal, which is confusing
+https://rust-lang-nursery.github.io/rust-clippy/master/index.html
```
Comment (mcarton) [inline README.md]: I suggest something like

```md
[There are 209 lints included in this crate!](https://rust-lang-nursery.github.io/rust-clippy/master/index.html)
```

which renders as

> [There are 209 lints included in this crate!](https://rust-lang-nursery.github.io/rust-clippy/master/index.html)

Rather that the current which renders as

> There are 209 lints included in this crate: https://rust-lang-nursery.github.io/rust-clippy/master/index.html

File: util/update_lints.py [hunk 3]
```diff
@@ -29,7 +29,7 @@
 
 nl_escape_re = re.compile(r'\\\n\s*')
 
-wiki_link = 'https://github.com/rust-lang-nursery/rust-clippy/wiki'
+wiki_link = 'https://rust-lang-nursery.github.io/rust-clippy/master/index.html'
```
Comment (killercup) [inline util/update_lints.py]: I'd rename this to `docs_link`

**Comment by killercup** [comment]
🎉 

Maybe also set https://rust-lang-nursery.github.io/rust-clippy/ as the homepage of the github repo?

**Comment by oli-obk** [comment]
> Maybe also set https://rust-lang-nursery.github.io/rust-clippy/ as the homepage of the github repo?

Do you mean of the crate? So it shows up on crates.io? Or what are homepages of github repos?

**Comment by killercup** [comment]
Ah, sorry, that was expressed quite poorly. Maybe a picture says more than a bunch of rambled words?

rust-lang-nursery_rust-clippy__a_bunch_of_lints_to_catch_common_mistakes_and_improve_your_rust_code

Setting it in the Cargo.toml (homepage or documentation?) might be a good idea, too.

**Comment by oli-obk** [comment]
@Manishearth you'd need to do that change

File: clippy_lints/src/utils/mod.rs (hunk 1 above)
Comment (mcarton) [reply clippy_lints/src/utils/mod.rs]: Github masked this comment but it is still valid.

**Comment by mcarton** [comment]
@killercup good idea
I can't seem to find how to do that, @Manishearth are you still the project's owner? You might have the right to do that.

**Comment by Manishearth** [comment]
Yeah, done. I don't think I have the ability to grant owner access anymore (just commit).

**Comment by oli-obk** [comment]
The address is wrong. It should be https://rust-lang-nursery.github.io/rust-clippy/

**Pull request merged**